logstash-core/lib/logstash/state_resolver.rb

```diff
@@ -41,14 +41,21 @@ def resolve(pipelines_registry, pipeline_configs)
         end
       end
 
-      configured_pipelines = pipeline_configs.map { |config| config.pipeline_id.to_sym }
+      configured_pipelines = pipeline_configs.each_with_object(Set.new) { |config, set|
+        set.add(config.pipeline_id.to_sym)
+      }
```

Comment:
style: we try to use brackets notation for blocks {|...| } for one-liners otherwise use do |...| end when multiline.